config/supermicro.go

```diff
@@ -114,13 +125,208 @@ func (cm *supermicroVendorConfig) Marshal() (string, error) {
 	}
 }
 
+func (cm *supermicroVendorConfig) Unmarshal(cfgData string) (err error) {
+	// the xml exported by sum is ISO-8859-1 encoded
+	decoder := xml.NewDecoder(bytes.NewReader([]byte(cfgData)))
+	// convert characters from non-UTF-8 to UTF-8
+	decoder.CharsetReader = charset.NewReaderLabel
+
+	err = decoder.Decode(cm.ConfigData.BiosCfg)
+	if err != nil {
+		return err
+	}
+
+	return
+}
+
+func (cm *supermicroVendorConfig) StandardConfig() (biosConfig map[string]string, err error) {
+	biosConfig = make(map[string]string)
+
+	for _, menu := range cm.ConfigData.BiosCfg.Menus {
+		for _, s := range menu.Settings {
+			switch s.Name {
+			// We want to drop this list of settings
+			case "NewSetupPassword", "NewSysPassword", "OldSetupPassword", "OldSysPassword":
+			// All others get normalized
+			default:
+				var k, v string
+				k, v, err = normalizeSetting(s)
+
+				if err != nil {
+					return
+				}
+
+				biosConfig[k] = v
+			}
+		}
+	}
+
+	return
+}
+
+func normalizeSetting(s *supermicroBiosCfgSetting) (k, v string, err error) {
```

Comment:
Test cases for this method would help maintain this over time

Reply:
Same comment re a future PR.